src/System/Process/Typed/Internal.hs

```diff
@@ -1,7 +1,702 @@
 {-# LANGUAGE CPP #-}
-module System.Process.Typed.Internal (
-  nullDevice
-) where
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveFunctor #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+-- | This module is __internal__ and its contents may change without a warning
+-- or announcement. It is not subject to the PVP.
+module System.Process.Typed.Internal
+    ( -- * Types
+      ProcessConfig(..)
+    , StreamSpec(..)
+    , StreamType (..)
+    , Cleanup(..)
+
+      -- * ProcessConfig
+      -- ** Smart constructors
+    , proc
+    , shell
+
+      -- | #processconfigsetters#
+
+      -- ** Setters
+    , setStdin
+    , setStdout
+    , setStderr
+    , setWorkingDir
+    , setWorkingDirInherit
+    , setEnv
+    , setEnvInherit
+    , setCloseFds
+    , setCreateGroup
+    , setDelegateCtlc
+#if MIN_VERSION_process(1, 3, 0)
+    , setDetachConsole
+    , setCreateNewConsole
+    , setNewSession
+#endif
+#if MIN_VERSION_process(1, 4, 0) && !WINDOWS
+    , setChildGroup
+    , setChildGroupInherit
+    , setChildUser
+    , setChildUserInherit
+#endif
+
+      -- | #streamspecs#
+
+      -- * Stream specs
+      -- ** Built-in stream specs
+    , inherit
+    , nullStream
+    , closed
+    , byteStringInput
+    , byteStringOutput
+    , byteStringFromHandle
+    , createPipe
+    , useHandleOpen
+    , useHandleClose
+
+      -- ** Create your own stream spec
+    , mkStreamSpec
+    , mkPipeStreamSpec
+
+      -- * Exceptions
+    , ExitCodeException (..)
+    , ByteStringOutputException (..)
+
+      -- * Helper functions
+    , nullDevice
+    , bracket
+    , finally
+    ) where
```

Comment:
Is there any point restricting the exports for the `Internal` module? As an `Internal` module I would expect it to export everything.

Reply:
No, there is no particular reason. It is simply a leftover from the refactoring; I'll remove the export list.

Reply:
Done.